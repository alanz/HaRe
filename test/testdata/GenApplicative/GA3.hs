module GA3 where

zipperM :: [a] -> IO [(a, Integer)]
zipperM lst = do
  lst2 <- getOtherList
  return $ zip lst lst2

getOtherList = return [1..]
