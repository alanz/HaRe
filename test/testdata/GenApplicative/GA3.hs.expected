module GA3 where

zipperM :: [a] -> IO [(a, Integer)]
zipperM lst = zip lst <$> getOtherList

getOtherList = return [1..]